File: container_with_most_water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left_ptr = 0
        right_ptr = len(height) - 1
        
        max_water = 0 
        while left_ptr < right_ptr:
            water_amount = (right_ptr - left_ptr) * min(height[left_ptr], height[right_ptr])
            max_water = max(max_water, water_amount)
            if height[left_ptr] < height[right_ptr]:
                left_ptr += 1
            else :
                right_ptr -= 1
                
        return max_water
            

File: container_with_most_water.java
import java.util.*;
class Solution {
    public int maxArea(int[] height) {
        int right=(height.length)-1;
        int left=0;
        int maxArea=0;
        int currArea;
        
        while(left<right){
            
            currArea=(right-left)* (Math.min(height[left], height[right]));
            maxArea=Math.max(currArea, maxArea);
            
            if( height[left] < height[right] ){
                left++; 
            }  
            
            else if( height[left] >= height[right] ){
                right--; 
            }  
        }
    return maxArea;       
        
        
    }
}
